pkg/messaging/mqtt/publisher.go

```diff
@@ -8,11 +8,10 @@ import (
 	"time"
 
 	mqtt "github.com/eclipse/paho.mqtt.golang"
+
 	"github.com/mainflux/mainflux/pkg/messaging"
 )
 
-var _ messaging.Publisher = (*publisher)(nil)
```

Comment:
Please, don't remove the interface check.

Reply:
It is useless because it is checked by the function `NewPublisher(address string, timeout time.Duration) (messaging.Publisher, error)`

Reply:
I agree that it is not necessary because of the factory function. However, it is in sync with the rest of the codebase, it is self-documenting, and it is an explicit type assertion, rather than an implicit conversion that happens in `NewPublisher`, so I would also prefer if we keep it. The same goes for `Subscriber`, but with the correct interface. Other than this, the PR looks good and ready for merge.